### DISCUSSION TITLE
Download from cat3movie.org not possible?

### DISCUSSION THREAD

**Discussion by Dynamix69** [aclap-dev/video-downloadhelper#1194] [Help and Q&A]
### Your question

Hi there,
on most websites VDH works well. But on this one [cat3movie.org ](https://cat3movie.org/) it just loads a few seconds and then it breaks up. 
Any Idea how to get these things downloaded?

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
{
    "used_history_button": true,
    "download_directory": "C:\\Users\\xxx\\Downloads\\Movies",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\xxx\\Downloads\\Movies"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 3,
    "successfull_dl": 54,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

One idea I have is for you to post at least one URL of a movie that you are having trouble downloading.  Don't say it's all of them.  I'm not going to investigate all of them.  I'm going to investigate any particular movies you tell me about, maybe 2 or 3, but at least one.  I'm not scrounging around on the site to find one.  Tell me about at least one & we'll take it from there.

**Reply by Dynamix69** [reply]
To be honest no Video is downloadable there anymore. Some months before it worked well.
Just to have a single example:  https://cat3movie.org/watch-the-woman-in-blue-1973/ep-full-server1.html

**Reply by GrampaWildWilly** [reply]
This one has me baffled.  When the page first loads & the player renders, you click the play button & it plays a 1 second video.  Then it shows the player as ready again.  When you click the play button a second time, the movie starts playing.  But nothing shows up in the Network Monitor.  That's why VDH doesn't recognize anything.  Well, VDH recognizes the 1 second video, which it is quite happy to download.  Fat lot of good that is.  There is a file shown in the Network Monitor with the name video.txt.  It is of type plain.  But if you look inside it, it looks like an  HLS stream manifest.  The problem is, each chunk of the stream is a png.  Movies are not made up of chunks of png files.  PNG files are images.  Still images.  Movies are made up of chunks of .ts files.  There is no DRM protection on this but it looks like the site has invented their own flavor of something that is acting just like DRM.  I've never been to this site before so I can't compare today's situation with what used to work some time ago.  It's entirely possible, although we have no concrete evidence of it, that the site changed their way of doing things.  I'm totally at a loss here.  Maybe @paulrouget Paul or @mattmjs Matt can see something here.

**Reply by Dynamix69** [reply]
Yes, that what I´ve seen as well: 1 second download, that´s it. I can enssure, that some Months ago, I was able to download from there, but that stopped some day and introduced this new behavior. I frist thought it´s a temporary problem (on my own) but when I came back some days ago and retried it, I ended up again in this situation. So it seem to me, that they truely have some sort of protection here.

**Reply by GrampaWildWilly** [reply]
I tried Matt's trick of copying the URL of that video.txt object & changing its name from .txt to .m3u8.  This is what I got:

[ffprobe.txt](https://github.com/user-attachments/files/16824255/ffprobe.txt)

It didn't tell me 404 File not found.  But those are way strange results.  A duration of about 96 minutes.  No audio track.  The video track is png.  Somehow, those files have a name of png but they are really ts chunks.  But the naming is blocking ffmpeg.

I tried copying the video.txt file to my system.  I renamed the file to be .m3u8.  To simplify things, the manifest actually has complete URLs in it, not partial URLs, so that saves a bit of hassle.  I mass changed all the .png extensions inside the file to .ts.  Then I tried to push that through ffmpeg.  Once again, I did not get 404 errors on the chunks.  But ffmpeg didn't actually do anything with them.  And eventually, the alleged download just failed.  It created a length 0 mp4.

I think the player on that page is transforming those png URLs into ts chunks as it plays the movie.  But none of that is showing up in the Network Monitor.  You can see the png chunks showing up in the Network Monitor.  These guys seem to have gone to great lengths to thwart downloading.

**Reply by Dynamix69** [reply]
If this example catches on, that's probably it for downloading. I'm not as deep into it as you are, but doesn't Netflix do something similar?

**Reply by GrampaWildWilly** [reply]
As I mentioned, this thing does not have DRM protection.  I test recorded a few seconds of this with OBS & it worked.  So you could use a screen recorder to record this stuff if you are totally desperate to have it.  Screen recording is suboptimal but at least it's available to you if you want it.

**Reply by GrampaWildWilly** [reply]
I have no experience with Netflix.  Never been a subscriber.

**Reply by GrampaWildWilly** [reply]
I believe Netflix straight up applies DRM protection.  So that's different.  But you'd need to get confirmation of that from a Netflix subscriber.

**Reply by Dynamix69** [reply]
I thought I´ve read the same behavior like you described for netflix somewhere. However Netflix can be recorded by using Audials for example. I think the same would work here. But for me it´s not worth to do such an overhead just to receive one or two movies I am intrested in. So I would prefer simply downloading it by something like VDH. If not possible, it´s not a big deal also. But intresting it is. ;-)

**Reply by Mattmjs** [reply]
Changing the .txt url to .m3u8 does make it detect the video but I can't get VDH or ffmpeg to download it.
I tried different file extension formats but all the same message. I even thought to try .png or .jpg which wouldn't make sense.

**yt-dlp** will work however download speed is around 230KB/s which quite slow. There is no quitting command like in ffmpeg
so I opened another command prompt to terminate the connection after around 5%.

**Reply by Dynamix69** [reply]
What are the settings then to download with yt-dlp? If I add just the url it´s saying that it´s not possible to download...

**Reply by Mattmjs** [reply]
**yt-dlp -o "name of the file.mp4" "url.m3u8"**

**Reply by Dynamix69** [reply]
So, this url.m3u8 is generated by renaming the video.txt file? If I try then this command I retrieve a lot of errors but no files...

**Reply by Mattmjs** [reply]
It's getting late maybe another time.

**Reply by Dynamix69** [reply]
No issue. Thanks a lot!

**Comment by Mattmjs** [comment]
I just thought I'd try out some other ideas, I changed the user agent to Firefox on android and it came up as m3u8 as opposed to txt.
It still doesn't work though. I tried in VLC with the m3u8 url, but VLC comes up empty handed as well.
How does it stream video in an image format which is the big mystery, probably this method is why ffmpeg is stumped.
The other thing is that the site has a devtools dectector which if you open the devtools while trying to play the video, 
the video will stop straight away thereby not show the stream information. Unless you have that setting enabled in Ublock Origin.

**Reply by Dynamix69** [reply]
So it seem that the point for this round goes to cat3movie. ;-)
However I did not manage to download via yt-dlp either. So I am still wondering how you did manage this (even it was slow).

**Reply by Mattmjs** [reply]
@Dynamix69 ,
My simple command I already posted is how it's done, you have to run the command prompt in the directory yt-dlp is in.
Then you input what I shown. I don't know what your doing wrong if you don't want the help. I've offered it more than
once but you don't respond. So maybe show some screenshots if interested otherwise I can't be bothered on it anymore.
